core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala

```diff
@@ -70,6 +70,10 @@ private[security] class HBaseDelegationTokenProvider
         getMethod("create", classOf[Configuration])
       confCreate.invoke(null, conf).asInstanceOf[Configuration]
     } catch {
+      case e: ClassNotFoundException =>
+        logWarning(s"You are attempting to use the ${getClass.getCanonicalName}, but" +
```

Comment:
The message is better than before but still only half correct, because end users don't intend to enable HBase delegation token provider.

I agree the warn message Spark provides is misleading so better to address that, but #23449 tackles root issue so for master branch (which gets rid of need for dealing with this confused message) I would rely on #23449 until the patch has a chance to be merged.

Reply:
@HeartSaVioR thanks for this suggestion, same opinion.